common/reactive/src/main/java/io/helidon/common/reactive/Collector.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.common.reactive;
+
+/**
+ * A collector accumulates the items provided when {@link #collect(java.lang.Object)} is invoked and makes them available in a
+ * single container object with {@link #value()}.
+ *
+ * @param <T> collected items type (input)
+ * @param <U> result container type (output)
+ */
+public interface Collector<T, U> {
+
+    /**
+     * Collect the given item.
+     *
+     * @param item item to collect
+     */
+    void collect(T item);
+
+    /**
+     * Get the collected items container.
+     *
+     * @return T
```

Comment:
typo? `@return T` should probably be `@return U`

Reply:
Good catch, will fix.

Reply:
Fixed.